File: csharp/1018.binary-prefix-divisible-by-5.cs
/*
 * @lc app=leetcode id=1018 lang=csharp
 *
 * [1018] Binary Prefix Divisible By 5
 */

using System.Collections.Generic;

// @lc code=start
public class Solution {
    public IList<bool> PrefixesDivBy5(int[] A) {
        int curr = 0;
        var ret = new List<bool>(A.Length);
        foreach (int num in A) {
            curr = ((curr << 1) | num) % 5;
            ret.Add(curr == 0);
        }
        return ret;
    }
}
// @lc code=end

File: java/1018.binary-prefix-divisible-by-5.java
/*
 * @lc app=leetcode id=1018 lang=java
 *
 * [1018] Binary Prefix Divisible By 5
 */

import java.util.ArrayList;
import java.util.List;

// @lc code=start

class Solution {
    public List<Boolean> prefixesDivBy5(int[] nums) {
        List<Boolean> ret = new ArrayList<>();
        int n = 0;
        for (int num : nums) {
            n = ((n << 1) | num) % 5;
            ret.add(n == 0);
        }
        return ret;
    }
}
// @lc code=end

